src/Compiler/Driver/FxResolver.fs

```diff
@@ -897,58 +798,51 @@ type internal FxResolver
-                        | Some tfm -> tfm
-                        | _ -> getRunningDotNetFrameworkTfm ()
+            let runningTfm = tryGetRunningTfm
 
+            // Coreclr has mechanism for getting rid
+            //      System.Runtime.InteropServices.RuntimeInformation.RuntimeIdentifier
+            // On Desktop framework compile it using osplatform+processarch+
             // Computer valid dotnet-rids for this environment:
             //      https://docs.microsoft.com/en-us/dotnet/core/rid-catalog
             //
             // Where rid is: win, win-x64, win-x86, osx-x64, linux-x64 etc ...
             let runningRid =
-                let processArchitecture = RuntimeInformation.ProcessArchitecture
-
-                let baseRid =
-                    if RuntimeInformation.IsOSPlatform(OSPlatform.Windows) then
-                        "win"
-                    elif RuntimeInformation.IsOSPlatform(OSPlatform.OSX) then
-                        "osx"
+                let rid =
+                    if isRunningOnCoreClr then
+                        // Use reflection to get the value for rid
+                        let rinfoType: Type option =
+                            Option.ofObj (Type.GetType("System.Runtime.InteropServices.RuntimeInformation, System.Runtime.InteropServices.RuntimeInformation", false))
+
+                        let ridProperty: PropertyInfo option =
+                            match rinfoType with
+                            | None -> None
+                            | Some t -> Option.ofObj (t.GetProperty("RuntimeIdentifier", BindingFlags.Public  ||| BindingFlags.Static))
+                        match ridProperty with
+                        | None -> None
+                        | Some p -> Some (string (p.GetValue(null)))
                     else
-                        "linux"
+                        None
+                match rid with
+                | Some rid -> rid
+                | None ->
+                    let processArchitecture = RuntimeInformation.ProcessArchitecture
+                    let baseRid =
+                        if RuntimeInformation.IsOSPlatform(OSPlatform.Windows) then
+                            "win"
+                        elif RuntimeInformation.IsOSPlatform(OSPlatform.OSX) then
+                            "osx"
+                        else
+                            "linux"
 
-                match processArchitecture with
-                | Architecture.X64 -> baseRid + "-x64"
-                | Architecture.X86 -> baseRid + "-x86"
-                | Architecture.Arm64 -> baseRid + "-arm64"
-                | _ -> baseRid + "-arm"
+                    match processArchitecture with
+                    | Architecture.X64 -> baseRid + "-x64"
+                    | Architecture.X86 -> baseRid + "-x86"
+                    | Architecture.Arm64 -> baseRid + "-arm64"
+                    | _ -> baseRid + "-arm"
```

Comment:
This could be simplified (?) to something like:

```fsharp
if isRunningOnCoreClr then
    // Use reflection to get the value for rid
    Option.ofObj (Type.GetType("System.Runtime.InteropServices.RuntimeInformation, System.Runtime.InteropServices.RuntimeInformation", false))
else None
|> Option.bind (fun rinfoType -> Option.ofObj (rinfoType.GetProperty("RuntimeIdentifier", BindingFlags.Public  ||| BindingFlags.Static)))
|> Option.map (fun ridProperty -> string (ridProperty.GetValue(null)))
|> Option.defaultWith (fun () ->
    let processArchitecture = RuntimeInformation.ProcessArchitecture
    let baseRid =
        if RuntimeInformation.IsOSPlatform(OSPlatform.Windows) then
            "win"
        elif RuntimeInformation.IsOSPlatform(OSPlatform.OSX) then
            "osx"
        else
            "linux"

    match processArchitecture with
    | Architecture.X64 -> baseRid + "-x64"
    | Architecture.X86 -> baseRid + "-x86"
    | Architecture.Arm64 -> baseRid + "-arm64"
    | _ -> baseRid + "-arm")
```

Reply:
I'm an old fashioned kind of guy ... as well as old.